graylog2-web-interface/src/views/logic/TimeRange.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import type { TimeRange, RelativeTimeRange } from 'views/logic/queries/Query';
+import { isTypeRelativeWithStartOnly, isTypeRelativeWithEnd } from 'views/typeGuards/timeRange';
+import assertUnreachable from 'logic/assertUnreachable';
+
+type SearchRelativeRangeStartOnly = {
+  rangetype: 'relative',
+  relative: string,
+};
+
+type SearchRelativeRangeWithEnd = {
+  rangetype: 'relative',
+  from: string,
+  to?: string,
+};
+
+type SearchRelativeRange = SearchRelativeRangeStartOnly | SearchRelativeRangeWithEnd
+
+type SearchAbsoluteRange = {
+  rangetype: 'absolute',
+  from: string,
+  to: string,
+};
+
+type SearchKeywordRange = {
+  rangetype: 'keyword',
+  keyword: string,
+  timezone?: string,
+};
+
+export type SearchTimeRange = SearchAbsoluteRange | SearchRelativeRange | SearchKeywordRange;
+
+export const toSearchTimerange = (timerange: TimeRange): SearchTimeRange => {
```

Comment:
It is a nitpick, but let's unify the wording for time range by using `timeRange` instead of `timerange`, when refactoring time range related code.

Reply:
:white_check_mark: